polytope_server/common/datasource/fdb.py

```diff
@@ -110,16 +110,22 @@ def get_type(self):
         return self.type
 
     def archive(self, request):
+        import pyfdb
+
+        fdb = pyfdb.FDB()
 
         # could add a check that the request is a singular object (does not contain)
 
         # r = yaml.safe_load(request.user_request)
-        self.fdb.archive(self.input_data)
-        self.fdb.flush()
+        fdb.archive(self.input_data)
+        fdb.flush()
         self.output = None
         return True
 
     def retrieve(self, request):
+        import pyfdb
+
+        self.fdb = pyfdb.FDB()
```

Comment:
Why here are we assigning `self.fdb` and not before in the init?

Reply:
following the previous answer, why here do you assign?

Reply:
I need to have it survive during the call to result which is where the data is being read from FDB and transferred to the staging area. However, it does not seem to work as intended.